eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -154,6 +154,38 @@ public function getXmlForImages($versionNo, array $fieldIds)
         return $fieldLookup;
     }
 
+    public function getAllVersionsImageXmlForFieldId(int $fieldId): array
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select(
+                $this->connection->quoteIdentifier('id'),
+                $this->connection->quoteIdentifier('version'),
+                $this->connection->quoteIdentifier('data_text')
+            )
+            ->from($this->connection->quoteIdentifier('ezcontentobject_attribute'))
+            ->where(
+                $selectQuery->expr()->eq(
+                    $this->connection->quoteIdentifier('id'),
+                    ':fieldId'
+                )
+            )
+            ->setParameter(':fieldId', $fieldId, PDO::PARAM_INT)
+        ;
+
+        $statement = $selectQuery->execute();
+
+        $fieldLookup = [];
+        foreach ($statement->fetchAll(PDO::FETCH_ASSOC) as $row) {
```

Comment:
`fetchAll` is deprecated, please see if for the supported DBAL version this works:
```suggestion
        foreach ($statement->fetchAllAssociative() as $row) {
```